src/components/views/voip/PictureInPictureDragger.tsx

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2017, 2018 New Vector Ltd
+Copyright 2019, 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import React, { createRef } from 'react';
+import UIStore from '../../../stores/UIStore';
+import { IApp } from '../../../stores/WidgetStore';
+import { lerp } from '../../../utils/AnimationUtils';
+import { MarkedExecution } from '../../../utils/MarkedExecution';
+import { replaceableComponent } from '../../../utils/replaceableComponent';
+import { CallViewHeader } from './CallView/CallViewHeader';
+
+const PIP_VIEW_WIDTH = 336;
+const PIP_VIEW_HEIGHT = 232;
+
+const MOVING_AMT = 0.2;
+const SNAPPING_AMT = 0.05;
+
+const PADDING = {
+  top: 58,
+  bottom: 58,
+  left: 76,
+  right: 8,
+};
+
+interface IProps {
+  children: React.ReactNode
+  draggable: boolean
+  app?: IApp
+}
+
+interface IState {
+  // Position of the PictureInPictureDragger
+  translationX: number;
+  translationY: number;
+}
+
+/**
+ * PictureInPictureDragger shows a small version of CallView hovering over the UI in 'picture-in-picture'
+ * (PiP mode). It displays the call(s) which is *not* in the room the user is currently viewing.
+ */
+@replaceableComponent("views.voip.PictureInPictureDragger")
+export class PictureInPictureDragger extends React.Component<IProps, IState> {
+  private callViewWrapper = createRef<HTMLDivElement>();
```

Comment:
Same comment here, we should rename this to make it less tied to calls

Reply:
See above